packages/vscode-extension/src/project/project.ts

```diff
@@ -101,20 +142,83 @@ export class Project
 
     this.devtools = new Devtools();
     this.metro = new Metro(this.devtools, this);
-    this.toolsManager = new ToolsManager(this.devtools, this.eventEmitter);
     this.start(false, false);
     this.trySelectingInitialDevice();
     this.deviceManager.addListener("deviceRemoved", this.removeDeviceListener);
     this.isCachedBuildStale = false;
+    this.toolsManager = new ToolsManager(this.devtools, this.eventEmitter);
 
-    this.fileWatcher = watchProjectFiles(() => {
-      this.checkIfNativeChanged();
-    });
-    this.licenseUpdater = refreshTokenPeriodically();
-    this.licenseWatcher = watchLicenseTokenChange(async () => {
-      const hasActiveLicense = await this.hasActiveLicense();
-      this.eventEmitter.emit("licenseActivationChanged", hasActiveLicense);
-    });
+    this.disposables.push(this.toolsManager);
+    this.disposables.push(
+      watchProjectFiles(() => {
+        this.checkIfNativeChanged();
+      })
+    );
+    this.disposables.push(refreshTokenPeriodically());
+    this.disposables.push(
+      watchLicenseTokenChange(async () => {
+        const hasActiveLicense = await this.hasActiveLicense();
+        this.eventEmitter.emit("licenseActivationChanged", hasActiveLicense);
+      })
+    );
+
+    this.disposables.push(
+      workspace.onDidChangeConfiguration(async (event: ConfigurationChangeEvent) => {
+        if (event.affectsConfiguration("launch")) {
+          const config = getCurrentLaunchConfig();
+          const oldAppRoot = this.appRootFolder?.getAppRoot();
+          if (config.appRoot === oldAppRoot) {
+            return;
+          }
+          const newAppRoot = await this.setupAppRoot();
+
+          if (newAppRoot === undefined) {
+            window.showErrorMessage(
+              "Unable to find the new app root, after a change in launch configuration. Radon IDE might not work properly.",
+              "Dismiss"
+            );
+            return;
+          }
+        }
+      })
+    );
+
+    this.disposables.push(
+      this.appRootFolder.onChangeAppRoot((newAppRoot: string) => {
+        const oldDependencyManager = this.dependencyManager;
+        this.dependencyManager = new DependencyManager(newAppRoot);
+        oldDependencyManager?.dispose();
+
+        const oldLaunchConfig = this.launchConfig;
+        this.launchConfig = new LaunchConfigController(newAppRoot);
+        oldLaunchConfig?.dispose();
+
+        this.buildCache = new BuildCache(newAppRoot);
+
+        this.reload("reboot");
+      })
+    );
+  }
+
+  private async setupAppRoot() {
+    const appRoot = findAppRootFolder();
+    if (!appRoot) {
+      return;
+    }
+
+    this.appRootFolder.setAppRoot(appRoot);
```

Comment:
It doesn't seem like having the AppRootFolder class makes sense any longer. We just use it in project and when the app root changes we also only call a code that is defined in the project class.